sstables/storage.cc

```diff
@@ -54,6 +55,8 @@ class filesystem_storage final : public sstables::storage {
     future<> touch_temp_dir(const sstable& sst);
     future<> move(const sstable& sst, sstring new_dir, generation_type generation, delayed_commit_changes* delay) override;
```

Comment:
Not necessarily a leak, as we don't lose track of them.

In retrospect: whenever the `filesystem_storage` object is destructed, the new `std::unordered_map` field (= the cache) goes away, tearing down all the contained `file`s too. As I've learned since, upon destruction, `file` performs a best-effort close. IOW, the directories (file descriptors at the lowest level, I guess) stick around as long as the `filesystem_storage` object does.

Reply:
It's still a leak. If you CREATE TABLE / DROP TABLE in a loop, they will accumulate.